apps/api/src/campaign-report/campaign-report.controller.ts

```diff
@@ -0,0 +1,211 @@
+  @UseInterceptors(
+    FileFieldsInterceptor(
+      [
+        { name: 'photos', maxCount: 5 },
+        { name: 'documents', maxCount: 5 },
+      ],
+      {
+        limits: { fileSize: 1024 * 1024 * 10 },
+      },
+    ),
+  )
+  async create(
+    @Param('campaignId') campaignId: string,
+    @UploadedFiles() files: { photos?: Express.Multer.File[]; documents?: Express.Multer.File[] },
+    @Body() report: CreateReportDto,
+    @AuthenticatedUser() user: KeycloakTokenParsed,
+  ) {
+    const authorizedPerson = await this.userCanPerformReportAction(campaignId, user)
+    if (!authorizedPerson) {
+      throw new ForbiddenException('The user cannot modify the requested campaign')
+    }
+
+    const createReportDto: CreateReportDto = plainToInstance(CreateReportDto, { ...report })
+    const createdReportId = await this.campaignReportService.createReport(
+      createReportDto,
+      campaignId,
+      authorizedPerson.id,
+      files.photos ?? [],
+      files.documents ?? [],
+    )
+    return createdReportId
+  }
+
+  @Patch(':campaignId/reports/:reportId')
+  @UseInterceptors(
+    FileFieldsInterceptor(
+      [
+        { name: 'photos', maxCount: 5 },
+        { name: 'documents', maxCount: 5 },
+      ],
+      {
+        limits: { fileSize: 1024 * 1024 * 10 },
+      },
+    ),
+  )
+  async update(
+    @Param('campaignId') campaignId: string,
+    @Param('reportId') reportId: string,
+    @UploadedFiles() files: { photos?: Express.Multer.File[]; documents?: Express.Multer.File[] },
+    @Body() updatedReport: UpdateReportDto,
+    @AuthenticatedUser() user: KeycloakTokenParsed,
+  ) {
+    const authorizedPerson = await this.userCanPerformReportAction(campaignId, user)
+    if (!authorizedPerson) {
+      throw new ForbiddenException('The user cannot modify the requested campaign')
+    }
+
+    const campaignReports = await this.campaignReportService.getReports(campaignId)
+
+    if (!campaignReports.map((report) => report.id).includes(reportId)) {
+      throw new NotFoundException('The given report is not part of the selected campaign')
+    }
+
+    await this.campaignReportService.updateReport(
+      reportId,
+      authorizedPerson.id,
+      updatedReport,
+      files.photos ?? [],
+      files.documents ?? [],
+    )
+  }
+
+  @Delete(':campaignId/reports/:reportId')
+  async delete(
+    @Param('campaignId') campaignId: string,
+    @Param('reportId') reportId: string,
+    @AuthenticatedUser() user: KeycloakTokenParsed,
+  ) {
+    const authorizedPerson = await this.userCanPerformReportAction(campaignId, user)
+    if (!authorizedPerson) {
+      throw new ForbiddenException('The user cannot modify the requested campaign')
+    }
+
+    const campaignReports = await this.campaignReportService.getReports(campaignId)
+
+    if (!campaignReports.map((report) => report.id).includes(reportId)) {
+      throw new NotFoundException('The given report is not part of the selected campaign')
+    }
```

Comment:
it will be more optimal instead of getting the reports to add a where clause in the database query